src/pages/task/metadata/Stepback.tsx

```diff
@@ -0,0 +1,75 @@
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Body } from "@leafygreen-ui/typography";
+import Icon from "components/Icon";
+import { MetadataItem } from "components/MetadataCard";
+import { size } from "constants/tokens";
+import { TaskQuery } from "gql/generated/types";
+import { useBreakingTask } from "hooks/useBreakingTask";
+import { useLastPassingTask } from "hooks/useLastPassingTask";
+
+type Props = {
+  taskId: string;
+};
+
+/**
+ * Returns whether the task is in stepback.
+ * @param task The task to check if it is in stepback.
+ * @returns Whether the task is in stepback.
+ */
+export function inStepback(task: TaskQuery["task"]) {
+  // lastFailingStepbackTaskId is set only in the middle/last of stepback (not the first task).
+  const stepback =
+    task?.stepbackInfo?.lastFailingStepbackTaskId !== undefined &&
+    task?.stepbackInfo?.lastFailingStepbackTaskId !== "";
+
+  // nextStepbackTaskId is set when the next task in stepback in known, in the beginning
+  // of stepback, it is known right away. In the rest of stepback, it is not.
+  const beginningStepback =
+    task?.stepbackInfo?.nextStepbackTaskId !== undefined &&
+    task?.stepbackInfo?.nextStepbackTaskId !== "";
+
+  // If the task is in stepback or beginning stepback, it is counted as in stepback.
+  return stepback || beginningStepback;
+}
+
+export const Stepback: React.FC<Props> = ({ taskId }) => {
+  const { loading, task: breakingTask } = useBreakingTask(taskId);
+  const { task: lastPassingTask } = useLastPassingTask(taskId);
+
+  // The last stepback task has an undefined last passing task (it is passing itself).
+  const isLastStepbackTask = lastPassingTask === undefined;
+
+  // The stepback is finished if there is a breaking task or we are on the last stepback task.
+  const finished = breakingTask !== undefined || isLastStepbackTask;
+
+  return (
+    <MetadataItem>
+      Stepback:
+      <Tooltip
+        align="top"
+        justify="middle"
+        trigger={
+          <IconContainer>
+            <Icon glyph="InfoWithCircle" size="small" />
+          </IconContainer>
+        }
+        triggerEvent="hover"
+      >
+        <Body>
+          When Stepback is completed you can access the breaking commit via the
+          relevant commits dropdown.
+        </Body>
+      </Tooltip>{" "}
+      {loading && <Badge variant="lightgray">Loading</Badge>}
```

Comment:
Having `Loading` in a badge makes it seem like it's a state of stepback, rather than an indication that we're still querying for a response. Perhaps a [Skeleton Loader](https://www.mongodb.design/component/skeleton-loader/example/) or [Spinner](https://www.mongodb.design/component/loading-indicator/example/) could be a good replacement? (a skeleton loader is good for longer-term loading operations > 300ms, a spinner is good for shorter-term loading operations < 300ms)

Reply:
It's hard to guage how fast this might take. I based it on relevant commits since it uses the same queries, and guessed on average it would be a longer operation so I did the skeleton loader for the badge!